js/src/views/Status/NodeStatus/store.js

```diff
@@ -37,37 +41,65 @@ export default class StatusStore {
     this.api = api;
   }
 
-  @action setLongStatus ({ defaultExtraData, enode, netPort, rpcSettings }) {
-    transaction(() => {
-      this.defaultExtraData = defaultExtraData;
-      this.enode = enode;
-      this.netPort = netPort;
-      this.rpcSettings = rpcSettings;
-    });
-  }
-
-  @action setStatus ({ hashrate }) {
-    transaction(() => {
-      this.hashrate = hashrate;
-    });
+  @action setStatuses ({ chain, defaultExtraData, enode, netPeers, netPort, rpcSettings, hashrate }) {
+    this.chain = chain;
+    this.defaultExtraData = defaultExtraData;
+    this.enode = enode;
+    this.netPeers = netPeers;
+    this.netPort = netPort;
+    this.rpcSettings = rpcSettings;
+    this.hashrate = hashrate;
   }
 
   @action setMinerSettings ({ coinbase, extraData, gasFloorTarget, minGasPrice }) {
-    transaction(() => {
-      this.coinbase = coinbase;
-      this.extraData = extraData;
-      this.gasFloorTarget = gasFloorTarget;
-      this.minGasPrice = minGasPrice;
-    });
+    this.coinbase = coinbase;
+    this.extraData = extraData;
+    this.gasFloorTarget = gasFloorTarget;
+    this.minGasPrice = minGasPrice;
   }
 
   startPolling () {
-    this._pollStatus();
-    this._pollLongStatus();
+    this._startPolling();
   }
 
   stopPolling () {
-    Object.keys(this._timeoutIds).forEach((key) => clearTimeout(this._timeoutIds[key]));
+    if (this.subscriptionId) {
```

Comment:
Should wait for `subscriptionId`, otherwise if you `stopPolling` before the promises in `startPolling` are resolved you may end up with dangling subscription.